src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/DataFlowPass.vb

```diff
@@ -491,6 +496,46 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return slot
         End Function
 
+        ''' <summary>
+        ''' Descends through Rest fields of a tuple if "symbol" is an extended field
+        ''' As a result the "symbol" will be adjusted to be the field of the innermost tuple
+        ''' and a corresponding containingSlot is returned.
+        ''' Return value -1 indicates a failure which could happen for the following reasons
+        ''' a) Rest field does not exist, which could happen in rare error scenarios involving broken ValueTuple types
+        ''' b) Rest is not tracked already and forceSlotsToExist is false (otherwise we create slots on demand)
+        ''' </summary>
+        Private Function DescendThroughTupleRestFields(ByRef symbol As Symbol, containingSlot As Integer, forceContainingSlotsToExist As Boolean) As Integer
+            Dim fieldSymbol = TryCast(symbol, TupleFieldSymbol)
+            If fieldSymbol IsNot Nothing Then
+                Dim containingType As TypeSymbol = DirectCast(symbol.ContainingType, TupleTypeSymbol).UnderlyingNamedType
+
+                ' for tuple fields the variable identifier represents the underlying field
+                symbol = fieldSymbol.TupleUnderlyingField
+
+                ' descend through Rest fields
+                ' force corresponding slots if do not exist
+                While containingType <> symbol.ContainingType
+                    Dim restField = TryCast(containingType.GetMembers(TupleTypeSymbol.RestFieldName).FirstOrDefault(), FieldSymbol)
+                    If restField Is Nothing Then
+                        Return -1
+                    End If
+
+                    If forceContainingSlotsToExist Then
+                        containingSlot = GetOrCreateSlot(restField, containingSlot)
+                    Else
+                        If Not _variableSlot.TryGetValue(New VariableIdentifier(restField, containingSlot), containingSlot) Then
+                            Return -1
+                        End If
+                    End If
+
+                    containingType = restField.Type.GetTupleUnderlyingTypeOrSelf()
```

Comment:
Nope. The only change to the "symbol" is to "untuplify" it for consistency. That is - the local ID will contain underlying symbol regardless whether we started with an actual tuple field or with an underlying. - For the def analysis, such fields should have same IDs.

The rest of the method just finds the chain of "Rest" parents.


---
In reply to: [78856670](https://github.com/dotnet/roslyn/pull/13810#discussion_r78856670) [](ancestors = 78856670)